internal/docstore/driver/driver.go

```diff
@@ -71,3 +75,28 @@ type Mod struct {
 	FieldPath []string
 	Value     interface{}
 }
+
+// A Query defines a query operation to find documents within a collection based
+// on a set of requirements.
+type Query struct {
+	FieldPaths []string         // the selected field paths
+	Filters    []Filter         // a list of filter for the query
+	Limit      int              // the number of result in one query request
+	StartAt    interface{}      // define the starting point of the query
+	Iter       DocumentIterator // the iterator for Get action
+}
+
+// A Filter defines a filter expression used to filter the query result.
+type Filter struct {
+	Field string      // the field path to filter
+	Op    string      // the operation, supports =, >, >=, <, <=
```

Comment:
Consider making an enum for this? Not sure it's worth it.

Reply:
I don't think it's worth it.